src/search.ts

```diff
@@ -0,0 +1,247 @@
+        limit: this._limit,
+        offset: this._offset,
+      });
+    return await lingo.callAPI("GET", path, { qs: { query: params } });
+  }
+
+  // Query Type
+  /*-------------------------------------------------------------------------------*/
+  /**
+   * Restrict results to content such as notes, guides, and assets. (default)
+   * @returns The chainable search object
+   */
+  content(): Search {
+    this._queryType = "content";
+    return this;
+  }
+  /**
+   * Restrict results to kits
+   * @returns The chainable search object
+   */
+  kits(): Search {
+    this._queryType = "jump_to";
+    return this.ofType("kit");
+  }
+  /**
+   * Restrict results to sections
+   * @returns The chainable search object
+   */
+  sections(): Search {
+    this._queryType = "jump_to";
+    return this.ofType("section");
+  }
+
+  /**
+   * Restrict results to headings
+   * @returns The chainable search object
+   */
+  headings(): Search {
+    this._queryType = "jump_to";
+    return this.ofType("heading");
+  }
+
+  // Options
+  /*-------------------------------------------------------------------------------*/
+  /**
+   *
+   * @param sort A property to sort by (relevance, type, alpha or recent)
+   * @param reverse If the sort should be reversed
+   * @returns The chainable search object
+   */
+  sortBy(sort: Sort, reverse = false): Search {
+    if (sort == "relevance" && reverse) {
+      throw Error("Relevance sort cannot be reversed");
+    }
+    this._sort = reverse ? `-${sort}` : sort;
+    return this;
+  }
+
+  limit(value: number): Search {
+    this._limit = value;
+    return this;
+  }
+
+  offset(value: number): Search {
+    this._offset = value;
+    return this;
+  }
+
+  /**
+   * Adjust the offset to the next page based on the current limit
+   * @returns The chainable search object
+   */
+  nextPage(): Search {
+    return this.offset(this._offset + this._limit);
+  }
+
+  // Filters
+  /*-------------------------------------------------------------------------------*/
+
+  /**
+   * Restricts results to a give kit. If called multiple times the query will use an OR operator for all provided kits.
+   * @param id The id of a kit
+   * @param version The version of a kit to search (defaults to the recommended version)
+   * @returns The chainable search object
+   */
+  inKit(id: string, version = 0): Search {
+    this._filters.push({
+      type: "kit",
+      kit_uuid: id,
+      version,
+    });
+    return this;
+  }
+
+  /**
+   * * Restricts results to a given section. If called multiple times the query will use an OR operator for all provided sections.
+   * @param id The id of a section
+   * @param version The version of the section to search (defaults)
+   * @returns The chainable search object
+   */
+  // TODO:
+  /*
+  inSection(id: string, version = 0): Search {
+    this._filters.push({
+      type: "section",
+      value: id,
+      uuid: id,
+      version,
+    });
+    return this;
+  }
+  */
+
```

Comment:
Plan on completing this for this release?

Reply:
It's actually broken in the API right now so I just left it here.